File: src/main/web/src/app/shop/shared/model/event-type.ts
export enum EventType {
	merch = "merch",
	tours = "tours",
	partys = "partys"
}


export function integerToType(eventType: number) {
	switch (eventType) {
		case 3:
			return EventType.merch;
		case 2:
			return EventType.partys;
		case 1:
			return EventType.tours;
	}
	return 0;
}

export function typeToInteger(eventType: EventType) {
	switch (eventType) {
		case EventType.merch:
			return 3;
		case EventType.partys:
			return 2;
		case EventType.tours:
			return 1;
	}
	return 0;
}

export function getEventTypes(): EventType[] {
	return Object.keys(EventType).map(typeKey => EventType[typeKey]);
}

File: src/main/java/memo/util/model/EventType.java
package memo.util.model;

import java.util.Arrays;
import java.util.Optional;

public enum EventType {
    tours(1, "tours", "tour"),
    partys(2, "partys", "party"),
    merch(3, "merch", "merch");


    private int value;
    private String stringRepresentation;
    private String permissionKey;

    EventType(int value, String stringRepresentation, String permissionKey) {
        this.value = value;
        this.stringRepresentation = stringRepresentation;
        this.permissionKey = permissionKey;
    }

    public int getValue() {
        return value;
    }

    public String getStringRepresentation() {
        return stringRepresentation;
    }

    public String getPermissionKey() {
        return permissionKey;
    }

    public static Optional<EventType> findByValue(int value) {
        return Arrays.stream(EventType.values())
                .filter(type -> type.value == value)
                .findFirst();
    }

    public static Optional<EventType> findByString(String value) {
        return Arrays.stream(EventType.values())
                .filter(type -> type.stringRepresentation.equals(value))
                .findFirst();
    }
}
